benches/bench.rs

```diff
@@ -153,6 +175,12 @@ fn criterion_benchmark(c: &mut Criterion) {
         4096,
         8192
     ];
+
+    c.bench_function("power of two 16", benchmark_power_of_two::<16>);
+    c.bench_function("non power of two 16", benchmark_non_power_of_two::<16>);
+
+    c.bench_function("power of two 1024", benchmark_power_of_two::<1024>);
+    c.bench_function("non power of two 1024", benchmark_non_power_of_two::<1024>);
```

Comment:
Instead of adding new benchmarking functions for the non-power-of-two case, it may make more sense to just add new calls to `generate_benches!`.  It'll get tested more thoroughly too.